src/claim_signer/GetClaimSigner.test.ts

```diff
@@ -0,0 +1,280 @@
+      should: 'be included in the calculation of the id and should be not equal to the work id',
+      actual: id !== TheRaven.id,
+      expected: true,
+    })
+  }
+
+  {
+    const work1: Claim = makeClaim({
+      name: ravenClaim.name,
+      author: ravenClaim.author,
+    })
+
+    const work2: Claim = makeClaim({
+      author: ravenClaim.author,
+      name: ravenClaim.name,
+    })
+
+    const id1 = await getClaimId(work1).catch(getErrorMessage)
+    const id2 = await getClaimId(work2).catch(getErrorMessage)
+
+    assert({
+      given: 'two claims with disordered keys',
+      should: 'have the same claims id',
+      actual: id1 === id2,
+      expected: true,
+    })
+  }
+
+  {
+    const work1: Claim = makeClaim({
+      name: ravenClaim.name,
+      author: ravenClaim.author,
+    })
+
+    const work2: Claim = makeClaim({
+      author: ravenClaim.author,
+      nAME: ravenClaim.name,
+      dateCreated: ravenClaim.dateCreated,
+      datePublished: ravenClaim.datePublished,
+    })
+
+    const id1 = await getClaimId(work1).catch(getErrorMessage)
+    const id2 = await getClaimId(work2).catch(getErrorMessage)
+
+    assert({
+      given: 'two claims with keys casing',
+      should: 'NOT have the same claims id',
+      actual: id1 !== id2,
+      expected: true,
+    })
+  }
+
+  {
+    const workClaim = makeClaim({
+      name: ravenClaim.name,
+      author: ravenClaim.author,
+    })
+
+    const TheRavenId = '3129d8056c04a00d3a84beaf38eed8aa25e2b7296ac08f8881a67c5cfcb1525e'
+
+    assert({
+      given: 'A work claim',
+      should: 'generate an id for the claim',
+      actual: await getClaimId(workClaim),
+      expected: TheRavenId,
+    })
+
+    const identityClaim = makeClaim({ ...Me.claim })
+
+    assert({
+      given: 'an identity claim',
+      should: 'generate an identityClaim for the claim',
+      actual: await getClaimId(identityClaim),
+      expected: 'bcc2843e20994e8c686c99fb92cd2feb0f3ab8c69e79e09e41f83635a6cd7fb9',
+    })
+
+    const workClaim2 = makeClaim({
+      name: ravenBookClaim.name,
+      author: ravenBookClaim.author,
+      isbn: ravenBookClaim.isbn,
+      edition: ravenBookClaim.edition,
+    })
+
+    assert({
+      given: 'an extended work claim WITHOUT proper context',
+      should: 'generate the id without the extra attributes',
+      actual: await getClaimId(workClaim2),
+      expected: TheRavenId,
+    })
+
+    assert({
+      given: 'an extended work claim WITH proper context',
+      should: 'generate a different id',
+      actual: (await getClaimId({ ...workClaim2, '@context': externalContext })) !== TheRavenId,
+      expected: true,
+    })
+  }
+})
+
+describe('createClaim', async (should: any) => {
+  const { assert } = should('')
+
+  {
+    const workClaim: any = await createClaim(testPrivateKeyEd25519Base58, ClaimType.Work, TheRaven.claim)
+
+    assert({
+      given: 'the public key of a Claim',
```

Comment:
is this test testing for `issuerId` or public key? are they the same? if so, is that a universal fact or an implementation detail?